docs: Clarify how to configure plain login (#1598)
redpanda-data/documentation: docs/reference/docker-compose.mdx

This `docker-compose.yml` file starts a single Redpanda broker, Redpanda Console, and a sample application called _owl shop_. Owl shop simulates a simple ecommerce shop that uses Redpanda as an asynchronous communication exchange. Owl shop creates topics, produces sample data to those topics, and consumes from those topics.

1. Paste the following YAML into a `docker-compose.yml` file on your computer, and run `docker compose up -d` inside the same directory.

  :::note
  If you are running on an ARM-based device such as the Apple M1 chip, uncomment the `platform: 'linux/amd64'` lines.
  :::

  ```yaml title="docker-compose.yml"
  version: '3.7'
  name: redpanda-owl-shop
  networks:
    redpanda_network:
      driver: bridge
  volumes:
    redpanda: null
  services:
    redpanda:
      image: docker.redpanda.com/redpandadata/redpanda:vVAR::REDPANDA_VERSION_23_1
      command:
        - redpanda start
        - --smp 1
        - --overprovisioned
        - --kafka-addr internal://0.0.0.0:9092,external://0.0.0.0:19092
        # Address the broker advertises to clients that connect to the Kafka API.

>         # Use the internal addresses to connect to the Redpanda brokers'

Reviewer: ```suggestion
        # Use the internal addresses to connect to the Redpanda brokers
```

      depends_on:
        - redpanda

    connect:
      image: docker.redpanda.com/redpandadata/connectors:latest
      hostname: connect
      container_name: connect
      networks:
        - redpanda_network
      # platform: 'linux/amd64'
      depends_on:
        - redpanda
      ports:
        - "8083:8083"
      environment:
        CONNECT_CONFIGURATION: |
            key.converter=org.apache.kafka.connect.converters.ByteArrayConverter
            value.converter=org.apache.kafka.connect.converters.ByteArrayConverter
            group.id=connectors-cluster
            offset.storage.topic=_internal_connectors_offsets
            config.storage.topic=_internal_connectors_configs
            status.storage.topic=_internal_connectors_status
            config.storage.replication.factor=-1
            offset.storage.replication.factor=-1
            status.storage.replication.factor=-1
            offset.flush.interval.ms=1000
            producer.linger.ms=50
            producer.batch.size=131072
        CONNECT_BOOTSTRAP_SERVERS: redpanda:19092
        CONNECT_GC_LOG_ENABLED: "false"
        CONNECT_HEAP_OPTS: -Xms512M -Xmx512M
        CONNECT_LOG_LEVEL: info
  ```

> 1. Open Redpanda Console at [localhost:8080](http://localhost:8080) and go to **Topics** to see the Owl Shop topics.

Reviewer 2: Should this be number 2 instead of step 1?

Author: It’s just some markdown syntax to avoid out of order steps. It’ll be converted to a 2 during the build

        - redpanda start
        - --smp 1
        - --overprovisioned
        - --kafka-addr internal://0.0.0.0:9092,external://0.0.0.0:19092
        # Address the broker advertises to clients that connect to the Kafka API.

>         # Use the internal addresses to connect to the Redpanda brokers'

Reviewer: ```suggestion
        # Use the internal addresses to connect to the Redpanda brokers
```

        # from inside the same Docker network.

>         # Use the external addresses to connect to the Redpanda brokers'

Reviewer: ```suggestion
        # Use the external addresses to connect to the Redpanda brokers
```

        # from outside the Docker network.
        - --advertise-kafka-addr internal://redpanda:9092,external://localhost:19092
        - --pandaproxy-addr internal://0.0.0.0:8082,external://0.0.0.0:18082
        # Address the broker advertises to clients that connect to the HTTP Proxy.
        - --advertise-pandaproxy-addr internal://redpanda:8082,external://localhost:18082
        - --schema-registry-addr internal://0.0.0.0:8081,external://0.0.0.0:18081

>         # Redpanda brokers use the RPC API to communicate with eachother internally.

Reviewer: ```suggestion
        # Redpanda brokers use the RPC API to communicate with each other internally.
```
